backpack/extensions/module_extension.py

```diff
@@ -12,98 +24,156 @@ class ModuleExtension:
       needs to be propagated through the graph.
     """
 
-    def __init__(self, params=None):
-        """
-        Parameters
-        ----------
-        params: [str]
-            List of module parameters that need special treatment.
-            for each param `p` in the list, instances of the extended module `m`
-            need to have a field `m.p` and the class extending `ModuleExtension`
-            need to provide a method with the same signature as the `backprop`
-            method.
-            The result of this method will be saved in the savefield of `m.p`.
+    def __init__(self, params: List[str] = None):
+        """Initialization.
+
+        Args:
+            params: List of module parameters that need special treatment.
+                for each param `p` in the list, instances of the extended module `m`
+                need to have a field `m.p` and the class extending `ModuleExtension`
+                need to provide a method with the same signature as the `backprop`
+                method.
+                The result of this method will be saved in the savefield of `m.p`.
+
+        Raises:
+            NotImplementedError: if child class doesn't have a method for each parameter
         """
         if params is None:
             params = []
 
         self.__params = params
 
         for param in self.__params:
-            extFunc = getattr(self, param, None)
-            if extFunc is None:
+            if hasattr(self, param) is False:
                 raise NotImplementedError
```

Comment:
Provide error message

Reply:
Wrote an error message. Please check whether it contains the correct information.

Reply:
Confirm.